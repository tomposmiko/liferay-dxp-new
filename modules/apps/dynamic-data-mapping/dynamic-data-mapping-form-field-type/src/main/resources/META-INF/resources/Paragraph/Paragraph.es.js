/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '../FieldBase/FieldBase.es';

import './ParagraphRegister.soy.js';

import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './Paragraph.soy.js';

class Paragraph extends Component {}

Paragraph.STATE = {
	/**
	 * @default 'string'
	 * @instance
	 * @memberof Text
	 * @type {?(string|undefined)}
	 */

	dataType: Config.string().value('string'),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	fieldName: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	id: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	label: Config.string().value(''),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	name: Config.string().required(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	placeholder: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof FieldBase
	 * @type {?(bool|undefined)}
	 */

	repeatable: Config.bool(),

	/**
	 * @default false
	 * @instance
	 * @memberof Paragraph
	 * @type {?(bool|undefined)}
	 */

	required: Config.bool().value(false),

	/**
	 * @default true
	 * @instance
	 * @memberof Paragraph
	 * @type {?(bool|undefined)}
	 */

	showLabel: Config.bool().value(true),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(string|undefined)}
	 */

	spritemap: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Paragraph
	 * @type {?(object|undefined)}
	 */

	text: Config.object(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof FieldBase
	 * @type {?(string|undefined)}
	 */

	tooltip: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Text
	 * @type {?(string|undefined)}
	 */

	type: Config.string().value('paragraph')
};

Soy.register(Paragraph, templates);

export default Paragraph;
