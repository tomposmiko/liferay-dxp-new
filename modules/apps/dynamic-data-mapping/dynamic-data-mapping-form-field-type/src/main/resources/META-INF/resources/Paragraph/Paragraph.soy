{namespace Paragraph}

{template .render}
	{@param name: string}
	{@param? id: string}
	{@param? label: string}
	{@param? text: html}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? spritemap: string}
	{@param? tooltip: string}
	{@param? visible: bool}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{call .content}
				{param name: $name /}
				{param text: $text /}
			{/call}
		{/param}
		{param id: $id /}
		{param label: $label /}
		{param name: $name /}
		{param repeatable: $repeatable /}
		{param required: $required /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tooltip: $tooltip /}
		{param visible: $visible /}
	{/call}
{/template}

{template .content}
	{@param? name: string}
	{@param? text: html}

	<div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="{$name}">
		<div class="liferay-ddm-form-field-paragraph-text">{$text}</div>
	</div>
{/template}