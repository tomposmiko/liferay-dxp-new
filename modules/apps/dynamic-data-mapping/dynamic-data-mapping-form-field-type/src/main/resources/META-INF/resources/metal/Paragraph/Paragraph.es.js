import './ParagraphRegister.soy.js';
import 'dynamic-data-mapping-form-field-type/metal/FieldBase/index.es';
import Component from 'metal-component';
import Soy from 'metal-soy';
import templates from './Paragraph.soy.js';
import {Config} from 'metal-state';

class Paragraph extends Component {
	static STATE = {

		/**
		 * @default 'string'
		 * @instance
		 * @memberof Text
		 * @type {?(string|undefined)}
		 */

		dataType: Config.string().value('string'),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		fieldName: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		id: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		label: Config.string().value(''),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		name: Config.string().required(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		placeholder: Config.string(),

		/**
		 * @default false
		 * @instance
		 * @memberof Paragraph
		 * @type {?(bool|undefined)}
		 */

		required: Config.bool().value(false),

		/**
		 * @default false
		 * @instance
		 * @memberof Paragraph
		 * @type {?(bool|undefined)}
		 */

		visible: Config.bool().value(true),

		/**
		 * @default undefined
		 * @instance
		 * @memberof FieldBase
		 * @type {?(bool|undefined)}
		 */

		repeatable: Config.bool(),

		/**
		 * @default true
		 * @instance
		 * @memberof Paragraph
		 * @type {?(bool|undefined)}
		 */

		showLabel: Config.bool().value(true),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		spritemap: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Paragraph
		 * @type {?(string|undefined)}
		 */

		tip: Config.string(),

		/**
		 * @default undefined
		 * @instance
		 * @memberof Text
		 * @type {?(string|undefined)}
		 */

		type: Config.string().value('paragraph'),

		/**
		 * @default undefined
		 * @instance
		 * @memberof FieldBase
		 * @type {?(string|undefined)}
		 */

		tooltip: Config.string()
	};
}

Soy.register(Paragraph, templates);

export default Paragraph;