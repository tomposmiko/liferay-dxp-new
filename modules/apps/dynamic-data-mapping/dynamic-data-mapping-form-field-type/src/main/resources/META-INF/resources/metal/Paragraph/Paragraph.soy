{namespace Paragraph}

{template .render}
	{@param name: string}
	{@param? id: string}
	{@param? label: string}
	{@param? text: html}
	{@param? visible: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? spritemap: string}
	{@param? tip: string}
	{@param? tooltip: string}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{call .content}
				{param name: $name /}
				{param text: $text /}
				{param visible: $visible /}
			{/call}
		{/param}
		{param id: $id /}
		{param label: $label /}
		{param repeatable: $repeatable /}
		{param required: $required /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
		{param tooltip: $tooltip /}
	{/call}
{/template}

{template .content}
	{@param? name: string}
	{@param? text: html}
	{@param? visible: bool}

	<div class="form-group{$visible ? '' : ' hide'} liferay-ddm-form-field-paragraph" data-fieldname="{$name}">
		<div class="liferay-ddm-form-field-paragraph-text">{$text}</div>
	</div>
{/template}