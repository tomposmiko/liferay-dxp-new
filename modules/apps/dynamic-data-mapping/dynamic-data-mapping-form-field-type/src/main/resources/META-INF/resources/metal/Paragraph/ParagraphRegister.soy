{namespace ParagraphRegister}

{deltemplate PageRenderer.RegisterFieldType variant="'paragraph'"}
	{call Paragraph.render data="all" /}
{/deltemplate}