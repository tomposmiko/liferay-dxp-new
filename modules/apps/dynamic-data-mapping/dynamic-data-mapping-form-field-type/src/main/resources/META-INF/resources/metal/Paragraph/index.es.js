import Paragraph from './Paragraph.es';

export default Paragraph;