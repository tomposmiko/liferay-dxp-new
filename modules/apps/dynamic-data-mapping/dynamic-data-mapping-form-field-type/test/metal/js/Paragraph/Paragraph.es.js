import Paragraph from 'source/Paragraph/Paragraph.es';

let component;
const spritemap = 'icons.svg';

const defaultParagraphConfig = {
	name: 'textField',
	spritemap
};

describe(
	'Field Paragraph',
	() => {
		afterEach(
			() => {
				if (component) {
					component.dispose();
				}
			}
		);

		it(
			'should be readOnly',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						readOnly: true
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a helptext',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						tip: 'Type something'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have an id',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						id: 'ID'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a label',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						label: 'label'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a placeholder',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						placeholder: 'Placeholder'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should not be required',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						required: false
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render Label if showLabel is true',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						label: 'text',
						showLabel: true
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a spritemap',
			() => {
				component = new Paragraph(defaultParagraphConfig);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a value',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						value: 'value'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should have a key',
			() => {
				component = new Paragraph(
					{
						...defaultParagraphConfig,
						key: 'key'
					}
				);

				expect(component).toMatchSnapshot();
			}
		);
	}
);