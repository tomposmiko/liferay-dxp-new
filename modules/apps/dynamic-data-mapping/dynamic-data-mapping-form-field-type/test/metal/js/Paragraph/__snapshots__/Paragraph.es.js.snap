// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Paragraph should be readOnly 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a helptext 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
  <span class="form-text">Type something</span>
</div>
`;

exports[`Field Paragraph should have a key 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a label 1`] = `
<div class="form-group">
  <label>label </label>
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a placeholder 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a spritemap 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a value 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have an id 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should not be required 1`] = `
<div class="form-group">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should render Label if showLabel is true 1`] = `
<div class="form-group">
  <label>text </label>
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;
